implants/lib/eldritch/src/file/moveto_impl.rs

```diff
@@ -6,97 +6,93 @@ pub fn moveto(old: String, new: String) -> Result<()> {
+    if Path::new(&new).is_dir() {
         fs::remove_dir_all(new.clone())?;
-    } else if Path::new(&new.clone()).is_file() {
+    } else if Path::new(&new).is_file() {
         fs::remove_file(new.clone())?;
     }
 
     fs::rename(old, new)?;
     Ok(())
 }
 
-
 #[cfg(test)]
 mod tests {
     use super::*;
-    use tempfile::{NamedTempFile,tempdir};
-    use std::path::Path;
     use std::fs::File;
+    use std::path::Path;
+    use tempfile::{tempdir, NamedTempFile};
 
     #[test]
-    fn test_moveto_dir() -> anyhow::Result<()>{
+    fn test_moveto_dir() -> anyhow::Result<()> {
         let tmp_dir = tempdir()?;
-        let path_old = String::from(tmp_dir.path().to_str().unwrap()).clone();
+        let path_old = String::from(tmp_dir.path().to_str().unwrap());
         let tmp_dir_new = tempdir()?;
-        let path_new = String::from(tmp_dir_new.path().to_str().unwrap()).clone();
+        let path_new = String::from(tmp_dir_new.path().to_str().unwrap());
         tmp_dir_new.close()?;
 
-
         moveto(path_old.clone(), path_new.clone())?;
 
-        assert_eq!(Path::new(&path_old).is_dir(), false);
-        assert_eq!(Path::new(&path_new).is_dir(), true);
+        assert!(!Path::new(&path_old).is_dir());
+        assert!(Path::new(&path_new).is_dir());
         Ok(())
     }
     #[test]
-    fn test_moveto_file() -> anyhow::Result<()>{
+    fn test_moveto_file() -> anyhow::Result<()> {
         let tmp_file = NamedTempFile::new()?;
-        let path_old = String::from(tmp_file.path().to_str().unwrap()).clone();
+        let path_old = String::from(tmp_file.path().to_str().unwrap());
         let tmp_file_new = NamedTempFile::new()?;
-        let path_new = String::from(tmp_file_new.path().to_str().unwrap()).clone();
+        let path_new = String::from(tmp_file_new.path().to_str().unwrap());
         tmp_file_new.close()?;
         // Run our code
 
         moveto(path_old.clone(), path_new.clone())?;
 
-        assert_eq!(Path::new(&path_old).is_file(), false);
-        assert_eq!(Path::new(&path_new).is_file(), true);
+        assert!(!Path::new(&path_old).is_file());
+        assert!(Path::new(&path_new).is_file());
         Ok(())
     }
     #[test]
-    fn test_moveto_file_overwrites() -> anyhow::Result<()>{
+    fn test_moveto_file_overwrites() -> anyhow::Result<()> {
         let tmp_file = NamedTempFile::new()?;
-        let path_old = String::from(tmp_file.path().to_str().unwrap()).clone();
+        let path_old = String::from(tmp_file.path().to_str().unwrap());
         let tmp_file_new = NamedTempFile::new()?;
-        let path_new = String::from(tmp_file_new.path().to_str().unwrap()).clone();
+        let path_new = String::from(tmp_file_new.path().to_str().unwrap());
         // Run our code
 
         moveto(path_old.clone(), path_new.clone())?;
 
-        assert_eq!(Path::new(&path_old).is_file(), false);
-        assert_eq!(Path::new(&path_new).is_file(), true);
+        assert!(!Path::new(&path_old).is_file());
```

Comment:
Lets switch this to `.exists()` instead of `.is_file()` while were at it. I think that's more clear this is checking the file no longer exists.